packages/alfa-dom/src/node/element.ts

```diff
@@ -255,6 +256,31 @@ export class Element extends Node implements Slot, Slotable {
     return path;
   }
 
+  protected _structurallyEquals(value: unknown): value is this {
+    return (
+      value instanceof Element &&
+      super._structurallyEquals(value) &&
+      this.namespace.equals(value.namespace) &&
+      this.prefix.equals(value.prefix) &&
+      this.name === value.name &&
+      // order of the attributes is irrelevant but we can't rely on Set#equals which will just call item.equals
+      // we check mutual inclusion of attributes, disregarding order.
+      this._attributes.every((thisAttribute) =>
+        value._attributes.some((valueAttribute) =>
+          thisAttribute.equals(valueAttribute, true)
+        )
+      ) &&
+      value._attributes.every((valueAttribute) =>
+        this._attributes.some((thisAttribute) =>
+          thisAttribute.equals(valueAttribute, true)
+        )
+      ) &&
+      this.style.equals(value.style) &&
```

Comment:
We also need to implement structural equality for the CSS structures as elements with inline styles otherwise won't be considered structurally equal unless `value === this`.